lale/lib/lale/aggregate.py

```diff
@@ -12,16 +12,122 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import ast
+import importlib
+
+import pandas as pd
+
 import lale.docstrings
 import lale.operators
 
+try:
+    import pyspark.sql as pysql
+    from pyspark.sql.dataframe import DataFrame as spark_df
+
+    spark_installed = True
+
+except ImportError:
+    spark_installed = False
+
+
+def _is_pandas_df(df):
+    return isinstance(df, pd.core.groupby.generic.DataFrameGroupBy) or isinstance(
+        df, pd.DataFrame
+    )
+
+
+def _is_spark_df(df):
+    if spark_installed:
+        return isinstance(df, pysql.GroupedData) or isinstance(df, spark_df)  # type: ignore
+
+
+def _is_ast_subscript(expr):
+    return isinstance(expr, ast.Subscript)
+
+
+def _is_ast_attribute(expr):
+    return isinstance(expr, ast.Attribute)
+
 
 class _AggregateImpl:
-    def __init__(self, columns, group_by=[]):
-        self._hyperparams = {"columns": columns, "group_by": group_by}
+    def __init__(self, columns=None, group_by=None):
```

Comment:
I would suggest keeping the original defaults: 
`def __init__(self, columns, group_by=[]):`

columns is required for us to perform aggregate.